src/Workspaces/Core/Portable/Diagnostics/DefaultAnalyzerAssemblyLoaderService.cs

```diff
@@ -1,44 +0,0 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
-
-using System;
-using System.Composition;
-using System.IO;
-using Microsoft.CodeAnalysis.Host.Mef;
-using System.Collections.Immutable;
-using System.Collections.Generic;
-
-namespace Microsoft.CodeAnalysis.Host;
-
-[ExportWorkspaceServiceFactory(typeof(IAnalyzerAssemblyLoaderProvider)), Shared]
-[method: ImportingConstructor]
-[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-internal sealed class DefaultAnalyzerAssemblyLoaderServiceFactory(
-    [ImportMany] IEnumerable<IAnalyzerAssemblyResolver> externalResolvers) : IWorkspaceServiceFactory
-{
-    public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
-        => new DefaultAnalyzerAssemblyLoaderProvider(workspaceServices.Workspace.Kind ?? "default", [.. externalResolvers]);
-
-    private sealed class DefaultAnalyzerAssemblyLoaderProvider(string workspaceKind, ImmutableArray<IAnalyzerAssemblyResolver> externalResolvers)
-        : IAnalyzerAssemblyLoaderProvider
-    {
-        /// <summary>
-        /// We include the <see cref="WorkspaceKind"/> of the workspace in the path we produce.  That way we don't
-        /// collide in the common case of a normal host workspace and OOP workspace running together.  This avoids an
-        /// annoying exception as each will try to clean up this directory, throwing exceptions because the other is
-        /// locking it.  The exception is fine, since the cleanup is just hygienic and isn't intended to be needed for
-        /// correctness.  But it is annoying and does cause noise in our perf test harness.
```

Comment:
As an observation -- this code is relatively new. I don't understand why it ever would have been necessary, but we should be on the lookout for this problem coming back. Maybe give it a few RPS runs first or something?